tests/js/spec/components/smartSearchBar/index.spec.jsx

```diff
@@ -1445,4 +1446,182 @@ describe('SmartSearchBar', function () {
       expect(searchBar.text()).not.toContain("isn't supported here");
     });
   });
+
+  describe('date fields', () => {
+    const props = {
+      query: '',
+      organization,
+      location,
+      supportedTags,
+    };
+
+    it('displays date picker dropdown when appropriate', () => {
+      render(<SmartSearchBar {...props} />);
+
+      const textbox = screen.getByRole('textbox');
+      userEvent.click(textbox);
+      expect(screen.queryByTestId('search-bar-date-picker')).not.toBeInTheDocument();
+
+      // Just lastSeen: will display relative and absolute options, not the datepicker
+      userEvent.type(textbox, 'lastSeen:');
+      expect(screen.queryByTestId('search-bar-date-picker')).not.toBeInTheDocument();
+      expect(screen.getByText('Last hour')).toBeInTheDocument();
+      expect(screen.getByText('After a custom datetime')).toBeInTheDocument();
+
+      // lastSeen:> should open the date picker
+      userEvent.type(textbox, '>');
+      expect(screen.getByTestId('search-bar-date-picker')).toBeInTheDocument();
+
+      // Continues to display with date typed out
+      userEvent.type(textbox, '2022-01-01');
+      expect(screen.getByTestId('search-bar-date-picker')).toBeInTheDocument();
+
+      // Goes away when on next term
+      userEvent.type(textbox, ' ');
+      expect(screen.queryByTestId('search-bar-date-picker')).not.toBeInTheDocument();
+
+      // Pops back up when cursor is back in date token
+      userEvent.keyboard('{arrowleft}');
+      expect(screen.getByTestId('search-bar-date-picker')).toBeInTheDocument();
+
+      // Moving cursor inside the `lastSeen` token hides the date picker
+      textbox.setSelectionRange(1, 1);
+      userEvent.click(textbox);
+      expect(screen.queryByTestId('search-bar-date-picker')).not.toBeInTheDocument();
+    });
+
+    it('can select a suggested relative time value', () => {
+      render(<SmartSearchBar {...props} />);
+
+      userEvent.type(screen.getByRole('textbox'), 'lastSeen:');
+
+      userEvent.click(screen.getByText('Last hour'));
+
+      expect(screen.getByRole('textbox')).toHaveValue('lastSeen:-1h ');
+    });
+
+    it('can select a specific date/time', async () => {
+      render(<SmartSearchBar {...props} />);
+
+      userEvent.type(screen.getByRole('textbox'), 'lastSeen:');
+
+      userEvent.click(screen.getByText('After a custom datetime'));
+
+      // Should have added '>' to query and show a date picker
+      expect(screen.getByRole('textbox')).toHaveValue('lastSeen:>');
+      expect(screen.getByTestId('search-bar-date-picker')).toBeInTheDocument();
+
+      // Select a day on the calendar
+      await tick();
```

Comment:
For whatever reason, need this `await tick()` to get the lazily-loaded datepicker to show up in this test. Without it, the datepicker never shows up, no matter how long I set the timeout in `findByTestId`.

Interestingly, this only happens due to how large this test file is. If I skip enough of the other tests, it loads fine. Anyone have any idea why this might be happening? 🤔

Reply:
some weird global state thing is definitely going on, would need to really dig into how it gets loaded 😢

Reply:
leave this as a comment here I think